offscreen_render_target/offscreen_render_target.cpp

```diff
@@ -315,21 +334,23 @@ namespace bnb::oep
                 return nullptr;
         }
 
-        size_t size = m_width * m_height * 4;
+        size_t size = width * height * 4;
         auto plane_storage = std::shared_ptr<uint8_t>(new uint8_t[size]);
-        bnb::oep::interfaces::pixel_buffer::plane_data bpc8_plane{plane_storage, size, m_width * 4};
+        bnb::oep::interfaces::pixel_buffer::plane_data bpc8_plane{plane_storage, size, width * 4};
 
         glBindFramebuffer(GL_FRAMEBUFFER, m_last_framebuffer);
-        GL_CALL(glReadPixels(0, 0, m_width, m_height, gl_format, GL_UNSIGNED_BYTE, plane_storage.get()));
+        GL_CALL(glReadPixels(0, 0, width, height, gl_format, GL_UNSIGNED_BYTE, plane_storage.get()));
         glBindFramebuffer(GL_FRAMEBUFFER, 0);
 
         std::vector<bnb::oep::interfaces::pixel_buffer::plane_data> planes{bpc8_plane};
-        return bnb::oep::interfaces::pixel_buffer::create(planes, format_hint, m_width, m_height);
+        return bnb::oep::interfaces::pixel_buffer::create(planes, format_hint, width, height);
     }
 
     /* offscreen_render_target::read_current_buffer_i420 */
     pixel_buffer_sptr offscreen_render_target::read_current_buffer_i420(bnb::oep::interfaces::image_format format_hint)
     {
+        int32_t width = m_swap_sizes ? m_height : m_width;
+        int32_t height = m_swap_sizes ? m_width : m_height;
```

Comment:
copy-paste. It is bad. But as a temporary solution it's good
 Refactoring needed